TreeMaker/src/TreeMaker.cc

```diff
@@ -48,10 +48,32 @@ TreeMaker::TreeMaker(const edm::ParameterSet& iConfig) :
 	treeName = iConfig.getParameter<string>("TreeName");
 	doLorentz = iConfig.getParameter<bool>("doLorentz");
 	sortBranches = iConfig.getParameter<bool>("sortBranches");
-	//loop over all var type names to initialize TreeObjects
-	set<string> nameSet;
+
+	// parse the TitleMap
 	stringstream skipMessage;
 	stringstream message;
+	set<string> titleSet;
+	vector<string> titleVector = iConfig.getParameter< vector<string> >("TitleMap");
+	message << "TitleMap:\n";
+	for(unsigned i = 0; i < titleVector.size(); i+=2){
+		string variable = titleVector.at(i);
+		string title = titleVector.at(i+1);
+		message << "full name:" << variable << " -> title: " << title << "\n";
+		//check for an exact repeat of an existing name
+		//otherwise add an entry to the titleMap
+		if (titleSet.find(variable)!=titleSet.end()){
+			if(skipMessage.str().empty()) skipMessage << "Skipping repeated (branch,title) pairs:\n";
```

Comment:
technically this doesn't check for a repeated pair, but just a repeated key (branch name)

Reply:
also, prepending `"Skipping repeated branches:\n"` to `skipMessage` on L119 no longer makes sense, should probably take a similar approach down below

Reply:
I tried to fix both the branch/title message and the printing of the original skipMessage. Let me know what you think.

Reply:
Looks fine to me now, thanks